tests/pass/miri-alloc.rs

```diff
@@ -0,0 +1,12 @@
+extern "Rust" {
+    fn miri_alloc(size: usize, align: usize) -> *mut u8;
+    fn miri_dealloc(ptr: *mut u8, size: usize, align: usize);
+}
+
+fn main() {
+    unsafe {
+        let ptr = miri_alloc(123, 1);
+        std::ptr::write_bytes(ptr, 0, 123);
+        miri_dealloc(ptr, 123, 1);
+    }
+}
```

Comment:
To make sure this actually works without a global allocator, add `#![no_std]` to the test and use `core::ptr::write_bytes`.